Tests/BaseObjectTests.swift

```diff
@@ -16,181 +18,183 @@ class BaseObjectTests: QuickSpec {
+                    it("doesn't execute callback function when it is removed") {
+                        let listenId = baseObject.once(eventName, callback: callback)
+                        baseObject.off(listenId)
+                        baseObject.trigger(eventName)
 
-                    expect(callbackWasCalled) == false
+                        expect(callbackWasCalled) == false
+                    }
                 }
-            }
 
-            describe("listenTo") {
-                it("Should fire callback for an event on a given context object") {
-                    let contextObject = BaseObject()
+                describe("listenTo") {
+                    it("executes callback function for an event on a given context object") {
+                        let anotherBaseObject = ConcreteBaseObject()
 
-                    baseObject.listenTo(contextObject, eventName: eventName, callback: callback)
-                    contextObject.trigger(eventName)
+                        baseObject.listenTo(anotherBaseObject, eventName: eventName, callback: callback)
+                        anotherBaseObject.trigger(eventName)
 
-                    expect(callbackWasCalled) == true
+                        expect(callbackWasCalled) == true
+                    }
                 }
-            }
 
-            describe("listenToOnce") {
-                it("Should fire callback just one time for an event on a given context object") {
-                    let contextObject = BaseObject()
+                describe("listenToOnce") {
+                    it("executes callback function just one time for an event on a given context object") {
+                        let anotherBaseObject = ConcreteBaseObject()
 
-                    baseObject.listenToOnce(contextObject, eventName: eventName, callback: callback)
-                    contextObject.trigger(eventName)
+                        baseObject.listenToOnce(anotherBaseObject, eventName: eventName, callback: callback)
+                        anotherBaseObject.trigger(eventName)
 
-                    expect(callbackWasCalled) == true
+                        expect(callbackWasCalled) == true
 
-                    callbackWasCalled = false
-                    contextObject.trigger(eventName)
+                        callbackWasCalled = false
+                        anotherBaseObject.trigger(eventName)
 
-                    expect(callbackWasCalled) == false
+                        expect(callbackWasCalled) == false
+                    }
                 }
-            }
 
-            describe("off") {
-                it("Callback should not be called if removed") {
-                    let listenId = baseObject.on(eventName, callback: callback)
-                    baseObject.off(listenId)
-                    baseObject.trigger(eventName)
+                describe("off") {
+                    it("doesn't executes callback function if it is removed") {
```

Comment:
This ``it`` seems that is holding a context logic, do you think that is necessary to split into a ``context`` > ``it`` ?
It could be:
```swift
context("#off") {
    context("When it is removed") {
        it("doesn't execute callback") {}
    }
}
```